File: 2018/src/main/java/nl/remcoder/adventofcode/day1/Part2.java
package nl.remcoder.adventofcode.day1;

import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.HashSet;
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

public class Part2 {
    public static void main(String[] args) throws Exception {
        List<Integer> frequencyShifts = Files.lines(Paths.get(ClassLoader.getSystemResource("day1/input").toURI()))
                                     .map(Integer::parseInt)
                                     .collect(Collectors.toList());

        int frequency = 0;

        Set<Integer> frequencies = new HashSet<>();

        int frequencyCounter = 0;

        while (frequencies.add(frequency)) {
            frequency += frequencyShifts.get(frequencyCounter);

            frequencyCounter++;
            if (frequencyCounter >= frequencyShifts.size()) {
                frequencyCounter = 0;
            }
        }

        System.out.println(frequency);
    }
}

File: 2018/src/main/java/nl/remcoder/adventofcode/day1/Part2.kt
package nl.remcoder.adventofcode.day1

import java.nio.file.Files
import java.nio.file.Paths
import java.util.HashSet
import java.util.stream.Collectors

fun main(args: Array<String>) {
    val frequencyShifts = Files.lines(Paths.get(ClassLoader.getSystemResource("day1/input").toURI()))
            .map { Integer.parseInt(it) }
            .collect(Collectors.toList())

    var frequency = 0

    val frequencies = HashSet<Int>()

    var frequencyCounter = 0

    while (frequencies.add(frequency)) {
        frequency += frequencyShifts[frequencyCounter]

        frequencyCounter++
        if (frequencyCounter >= frequencyShifts.size) {
            frequencyCounter = 0
        }
    }

    println(frequency)
}